PowerValidatedSolutions.psm1

```diff
@@ -29873,6 +30001,81 @@ Function Remove-vRLIAlert {
 }
 Export-ModuleMember -Function Remove-vRLIAlert
 
+Function Set-vRLIAlert {
+    <#
+        .SYNOPSIS
+        Enable/Disable an alert
+
+        .DESCRIPTION
+        The Set-vRLIAlert cmdlet enables or disables an alert in vRealize Log Insight
+
+        .EXAMPLE
+        Set-vRLIAlert -id 0111952f-9aec-3872-b108-d70ec8a2981a -enabled true
+        This example enables the alert in vRealize Log Insight based on id provided
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$id,
+        [Parameter (Mandatory = $true)] [ValidateSet('true','false')] [String]$enabled
+    )
+
+    Try {
+        $json = '{
+            "ids":["'+ $id +'"],
+            "enabled":'+ $enabled +'
+        }'
+        $uri = "https://$vrliAppliance/api/v2/alerts/batch-subscribe"
+        Invoke-RestMethod -Method 'PATCH' -Uri $uri -Headers $vrliHeaders -Body $json 
+    } Catch {
+        Write-Error $_.Exception.Message
+    }
+}
+Export-ModuleMember -Function Set-vRLIAlert
+
+Function Update-vRLIAlert {
+    <#
+        .SYNOPSIS
+        Update the configuration of an alert
+
+        .DESCRIPTION
+        The Update-vRLIAlert cmdlet updates the configuration of an existing alert in vRealize Log Insight
+
+        .EXAMPLE
+        Update-vRLIAlert -id 0111952f-9aec-3872-b108-d70ec8a2981a -email administrator@rainpole.io
+        This example adds a single email address to an alert in vRealize Log Insight
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$id,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$email
+    )
+
+    Try {
+        $json = '{
+            "recipients":
+                {
+                    "emails": [
+                        "'+ $email +'"
+                    ],
+                    "webhookIds":[
+
+                    ],
+                    "vrops":{"autoClearAlertAfterTimeout":false,
+                    "autoClearAlertsTimeoutMinutes":15,
+                    "vcopsCriticality":"none",
+                    "vcopsResourceKindKey":"",
+                    "vcopsResourceName":""
+                }
+            }
+        }'
+        $uri = "https://$vrliAppliance/api/v2/alerts/$id"
```

Comment:
Does this need a version check for v1 vs v2 like `Get-vRLIAlert`?

Reply:
It may do but I can't test it for an older version